web/src/api/utils/file.ts

```diff
@@ -13,18 +13,19 @@ export function fileToTokenizedFile(file: File): TokenizedFile {
 /**
  * Common filename prefix for a given list of files
  * @param files Files
+ * @param getPath Function to extract the path from the file
  * @returns Common prefix for all files.
  */
-export function commonFilenamePrefix(files: File[]): string {
+export function commonFilenamePrefix(files: File[], getPath: (f: File) => string): string {

```

Comment:
This change is not needed anymore if we only apply this function on the original file path.